lib/OpenQA/WebAPI/Controller/API/V1/Table.pm

```diff
@@ -263,6 +276,18 @@ sub destroy {
     my $entry_name;
 
     try {
+        # Tables used in a group configured in YAML must not be modified manually
+        my $parameter;
+        $parameter = 'product_id' if $table eq 'Products';
+        $parameter = 'machine_id' if $table eq 'Machines';
+        $parameter = 'test_suite_id' if $table eq 'TestSuites';
+        my $job_templates = $schema->resultset('JobTemplates')->search({$parameter => $self->param('id')});
+        my %groups;
+        while (my $job_template = $job_templates->next) {
+            $groups{$job_template->group->name} = 1 if $job_template->group->template;
+        }
+        die 'Group' . (scalar(%groups) > 1 ? 's' : '') . ' ' . join(', ', sort keys(%groups)) . " must be updated through the YAML template\n" if %groups;
+
```

Comment:
Please [DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself). You are calling the same code multiple times, please [extract method](http://refactoring.com/catalog/extractMethod.html).